Marlin/pinsDebug.h

```diff
@@ -724,75 +178,130 @@ static bool pwm_status(uint8_t pin) {
   SERIAL_PROTOCOLPGM("  ");
 } // pwm_status
 
+
+#define COM_UNSHIFTED(T,L) (TCCR##T##A & (_BV(COM##T##L##1) | _BV(COM##T##L##0)))
+
+#define COM_PRINT(N,Z)  do { \
+     char temp_string[] = STRINGIFY(Z); \
+     switch(temp_string[0]) { \
+      case 'A' : \
+        SERIAL_PROTOCOLPAIR("    COM" STRINGIFY(N) "" STRINGIFY(Z) ": ", (COM_UNSHIFTED(N,Z) >> 6)); \
+        break; \
+      case 'B' : \
+        SERIAL_PROTOCOLPAIR("    COM" STRINGIFY(N) "" STRINGIFY(Z) ": ", (COM_UNSHIFTED(N,Z) >> 4)); \
+        break; \
+      case 'C' : \
+        SERIAL_PROTOCOLPAIR("    COM" STRINGIFY(N) "" STRINGIFY(Z) ": ", (COM_UNSHIFTED(N,Z) >> 2)); \
+        break; \
+      default:  \
+        SERIAL_PROTOCOLPAIR("  com_make L:",  temp_string); \
+        SERIAL_PROTOCOLPGM("spacer");\
+        for (uint8_t y = 0; y < 28; y++) {     \
+          if (temp_string[y] != 0) MYSERIAL.write(temp_string[y]);\
+          else break;\
+        }\
+        break; \
+    } \
+  } while(0);
+
+
+
+#define ICR_VALUE(N)  (ICR##N)
+#define CS_MAKE(N) (TCCR##N##B & (_BV(CS##N##0) | _BV(CS##N##1) | _BV(CS##N##2)) )
 #define WGM_MAKE3(N) (((TCCR##N##B & _BV(WGM##N##2)) >> 1) | (TCCR##N##A & (_BV(WGM##N##0) | _BV(WGM##N##1))))
 #define WGM_MAKE4(N) (WGM_MAKE3(N) | (TCCR##N##B & _BV(WGM##N##3)) >> 1)
 #define TIMER_PREFIX(T,L,N) do{ \
```

Comment:
How did you envision replacing all the name creation in this macro?

As best I can tell I'll need to create arrays of pointers to the registers something like this:
```
OCR_array[][A-C] PROGMEM = {

   #if defined(TCCR0A) && defined(COM0A1)
      #if defined (TIMER0A)
        {&OCR0A,&OCR0B,0},
      #else
        {0,&OCR0B,0},
      #endif
    #else  
        {0,0,0},
    #endif    

    #if defined(TCCR1A) && defined(COM1A1)
     #if defined(COM1C1) && defined (TIMER1C)
       {&OCR1A,&OCR1B,&OCR1C},
     #else
       {&OCR1A,&OCR1B,0},
     #endif 
    #else
        {0,0,0},   
    #endif

    #if defined(TCCR2A) && defined(COM2A1)
      {&OCR2A,&OCR2B,0},
    #endif

    #if defined(TCCR3A) && defined(COM3A1)
      {&OCR3A,&OCR3B,0},
     #if defined(COM3C1)
      {&OCR3A,&OCR3B,&OCR3C},
     #endif
    #endif

    #ifdef TCCR4A
      {&OCR4A,&OCR4B,&OCR4C},
    #endif

    #if defined(TCCR5A) && defined(COM5A1)
      {&OCR5A,&OCR5B,&OCR5C},
    #endif
  } 
```